threadPool.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include <iterator>
+#include "threadPool.h"
+
+
+ThreadPool::ThreadPool(const size_t thread_count)
+{
+	threads.reserve(thread_count);
+	try
+	{
+		for (size_t i = 0; i < thread_count; i++)
+		{
+			threads.push_back(
+				std::thread(&ThreadPool::workThread, this));
+		}
+	}
+	catch (const std::exception& ex)
+	{
+		run = false;
+		std::cout << ex.what() << std::endl;
+		throw;
+	}
+}
+
+ThreadPool::~ThreadPool()
+{
+	while (workQueue.size() > 0) {}
```

Comment:
what is the sense in this while loop?

Reply:
we are waiting for the queue to be empty, if we do not do this we may get different types of errors or incorrect execution of the program